zookeeper/src/main/java/com/linecorp/armeria/server/zookeeper/CuratorXInstanceSpecBuilder.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server.zookeeper;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.UUID;
+
+import javax.annotation.Nullable;
+
+import org.apache.curator.x.discovery.ServiceInstance;
+import org.apache.curator.x.discovery.ServiceType;
+import org.apache.curator.x.discovery.UriSpec;
+
+/**
+ * Builds an {@link InstanceSpec} for
+ * <a href="https://curator.apache.org/curator-x-discovery/index.html">Curator-X-Discovery</a>..
+ */
+public final class CuratorXInstanceSpecBuilder {
+
+    private final String serviceName;
+    @Nullable
+    private String serviceId;
+    @Nullable
+    private String serviceAddress;
+    @Nullable
+    private Integer port;
+    @Nullable
+    private Integer sslPort;
+    private ServiceType serviceType = ServiceType.DYNAMIC;
+    @Nullable
+    private Object payload;
+
+    private UriSpec uriSpec = new UriSpec("{scheme}://{address}:{port}");
+
+    /**
+     * Creates a new instance.
+     */
+    CuratorXInstanceSpecBuilder(String serviceName) {
+        this.serviceName = requireNonNull(serviceName, "serviceName");
+    }
+
+    /**
+     * Sets the service address.
+     */
+    public CuratorXInstanceSpecBuilder serviceAddress(String serviceAddress) {
+        this.serviceAddress = requireNonNull(serviceAddress, "serviceAddress");
+        return this;
+    }
+
+    /**
+     * Sets the port.
+     */
+    public CuratorXInstanceSpecBuilder port(int port) {
+        checkArgument(port > 0, "port: %s (expected: > 0)", port);
+        this.port = port;
+        return this;
+    }
+
+    /**
+     * Sets the SSL port.
+     */
+    public CuratorXInstanceSpecBuilder sslPort(int sslPort) {
```

Comment:
ssl or tls?

Reply:
The word `ssl` is used by the Curator so I followed it.
Isn't it better to use the word what they are using?
For the Eureka we used secure port. 🤔

Reply:
`sslPort` SGTM